### PR TITLE
Add D400 pydocstyle check - Amazon provider only

### PR DESCRIPTION
**Pull request by vincbeck**
This PR is a follow up of #31135. It had been decided to split #31135 in multiple PRs to make it easier to review. This PR scope is Amazon Provider only.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/amazon/aws/hooks/athena.py [hunk 1]
```diff
@@ -153,7 +153,7 @@ def get_query_results(
     ) -> dict | None:
         """
         Fetch submitted athena query results. returns none if query is in intermediate state or
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/athena.py]:
```suggestion
        Fetch submitted athena query results.

        Returns none if query is in intermediate state or
```

File: airflow/providers/amazon/aws/hooks/athena.py [hunk 2]
```diff
@@ -188,7 +188,7 @@ def get_query_results_paginator(
...
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/athena.py:190]:
```suggestion
        Fetch submitted athena query results.

        Returns none if query is in intermediate state or
```

File: airflow/providers/amazon/aws/hooks/athena.py [hunk 3]
```diff
@@ -227,7 +227,7 @@ def poll_query_status(
     ) -> str | None:
         """
         Poll the status of submitted athena query until query state reaches final state.
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/athena.py:230]:
```suggestion
        Poll the status of submitted athena query until query state reaches final state.

```

File: airflow/providers/amazon/aws/hooks/base_aws.py [hunk 4]
```diff
@@ -584,7 +584,7 @@ def get_session(self, region_name: str | None = None, deferrable: bool = False)
     def _get_config(self, config: Config | None = None) -> Config:
         """
         No AWS Operators use the config argument to this method.
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/base_aws.py:587]:
```suggestion
        No AWS Operators use the config argument to this method.

```

File: airflow/providers/amazon/aws/hooks/batch_client.py [hunk 5]
```diff
@@ -231,7 +231,7 @@ def terminate_job(self, job_id: str, reason: str) -> dict:
     def check_job_success(self, job_id: str) -> bool:
         """
         Check the final status of the Batch job; return True if the job
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/batch_client.py]:
```suggestion
        Check the final status of the Batch job.

        Return True if the job
```

File: airflow/providers/amazon/aws/hooks/ecs.py
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/ecs.py:1]: Missed the periods in this file?

File: airflow/providers/amazon/aws/hooks/redshift_sql.py [hunk 6]
```diff
@@ -147,7 +147,7 @@ def get_table_primary_key(self, table: str, schema: str | None = "public") -> li
         Helper method that returns the table primary key
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/redshift_sql.py]:
```suggestion
        Helper method that returns the table primary key.

```

File: airflow/providers/amazon/aws/hooks/redshift_sql.py [hunk 7]
```diff
@@ -147,7 +147,7 @@ def get_table_primary_key(self, table: str, schema: str | None = "public") -> li
...
         :param table: Name of the target table
         :param schema: Name of the target schema, public by default
-        :return: Primary key columns list
+        :return: Primary key columns list.
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/redshift_sql.py]:
```suggestion
        :return: Primary key columns list
```

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 8]
```diff
@@ -175,7 +175,7 @@ def parse_s3_url(s3url: str) -> tuple[str, str]:
         """
         Parses the S3 Url into a bucket name and key.
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/s3.py:176]:
```suggestion
        Parses the S3 Url into a bucket name and key.

```

File: airflow/providers/amazon/aws/hooks/s3.py [hunk 9]
```diff
@@ -207,8 +207,8 @@ def get_s3_bucket_key(
     ) -> tuple[str, str]:
         """
         Get the S3 bucket name and key from either:
```
Comment (uranusjr) [inline airflow/providers/amazon/aws/hooks/s3.py]:
```suggestion
        Get the S3 bucket name and key.

        From either:
```

**Comment by vincbeck** [comment]
Thanks for reviewing @uranusjr. I addressed your suggestions

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
LGTM. @uranusjr ?

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
Has conflicts

**Comment by vincbeck** [comment]
All green :)

**Pull request merged**